.travis.yml

```diff
@@ -7,143 +7,86 @@ env:
     env:
     - COMPILER=g++-4.9
-    - CMAKE_EXTRA_FLAGS="-DRAJA_ENABLE_CUDA=ON"
-    - TRAVIS_BUILD_NVCC="YES"
-  - compiler: gcc-5-debug
-    addons:
-      apt:
-        sources:
-        - ubuntu-toolchain-r-test
-        packages:
-        - g++-5
+    - CMAKE_EXTRA_FLAGS="-DRAJA_ENABLE_CUDA=On"
+    - TRAVIS_INSTALL_COMPILER="nvcc"
+  - compiler: gcc-4.9-debug
+    addons: { apt: { sources: [ ubuntu-toolchain-r-test ] , packages: [ g++-4.9 ] } }
     env:
-    - COMPILER=g++-5
-    - WARN="-DRAJA_ENABLE_WARNINGS=On"
-    - CMAKE_EXTRA_FLAGS="-DCMAKE_BUILD_TYPE=Debug"
-  - compiler: clang-3.8-debug
-    addons:
-      apt:
-        sources:
-        - ubuntu-toolchain-r-test
-        packages:
-        - g++-5
+    - COMPILER=g++-4.9
+    - CMAKE_EXTRA_FLAGS="-DCMAKE_BUILD_TYPE=Debug  -DRAJA_ENABLE_COVERAGE=On"
+  - compiler: clang-3.9-debug
+    addons: { apt: { sources: [ ubuntu-toolchain-r-test ] , packages: [ g++-6 ] } }
     env:
-    - COMPILER=$HOME/llvm/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-14.04/bin/clang++
-    - LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/llvm/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-14.04/lib
-    - DOWNLOAD=http://releases.llvm.org/3.8.0/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-14.04.tar.xz
-    - WARN=""
+    - COMPILER=clang++
+    - LLVM_VERSION=3.9.1
     - CMAKE_EXTRA_FLAGS="-DCMAKE_BUILD_TYPE=Debug"
   - compiler: nvcc-debug
-    addons:
-      apt:
-        sources:
-        - ubuntu-toolchain-r-test
-        packages:
-        - g++-4.9
+    addons: { apt: { sources: [ ubuntu-toolchain-r-test ] , packages: [ g++-4.9 ] } }
     env:
     - COMPILER=g++-4.9
-    - CMAKE_EXTRA_FLAGS="-DRAJA_ENABLE_CUDA=ON -DCMAKE_BUILD_TYPE=Debug"
-    - TRAVIS_BUILD_NVCC="YES"
+    - CMAKE_EXTRA_FLAGS="-DCMAKE_BUILD_TYPE=Debug -DRAJA_ENABLE_CUDA=On"
+    - TRAVIS_INSTALL_COMPILER="nvcc"
 cache:
   directories:
   - $HOME/llvm
-  - /usr/local/cuda-8.0
 before_install:
 - sudo apt-get update -qq
-- if [[ -n "$DOWNLOAD" ]] ; then if [[ ! -e "$COMPILER" ]] ; then cd $HOME; curl -o
-  tarball.tar.xz $DOWNLOAD &&
-  cd $HOME/llvm &&
-  tar xf $HOME/tarball.tar.xz ; fi ;
-  fi
+- mkdir -p ${HOME}/download
+- if [[ -n "${LLVM_VERSION}" ]]; then export LLVM_PATH=${HOME}/llvm/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-14.04; fi
+- if [[ -n "${LLVM_VERSION}" ]]; then export PATH=${LLVM_PATH}/bin:${PATH}; fi
+- if [[ -n "${LLVM_VERSION}" ]]; then export LD_LIBRARY_PATH=${LLVM_PATH}/lib:${LD_LIBRARY_PATH}; fi
+- if [[ -n "${LLVM_VERSION}" ]]; then export DOWNLOAD_URL=http://releases.llvm.org/${LLVM_VERSION}/clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-14.04.tar.xz; fi
+- if [[ -n "${LLVM_VERSION}" ]]; then export TARFILE=${HOME}/download/llvm-${LLVM_VERSION}.tar.xz; fi
+- if [[ -n "${LLVM_VERSION}" ]]; then if ! [[ -d "${LLVM_PATH}" ]]; then if ! [[ -f ${TARFILE} ]]; then echo "curl -o ${TARFILE} ${DOWNLOAD_URL}"; curl -o ${TARFILE} ${DOWNLOAD_URL}; fi; cd ${HOME}/llvm ; tar xf ${TARFILE}; fi; fi
```

Comment:
I like the re-factor in terms of what it's doing, but why the repeated test rather than a multi-line yaml block?  I'm not saying this needs to be fixed, it certainly works, I'm wondering about the reasoning.